src/LazyOperations/ConvexHull.jl

```diff
@@ -33,12 +33,12 @@ julia> typeof(c)
 ConvexHull{Float64,Ball2{Float64,Array{Float64,1}},Ball2{Float64,Array{Float64,1}}}
 ```
 """
-struct ConvexHull{N<:Real, S1<:LazySet{N}, S2<:LazySet{N}} <: LazySet{N}
+struct ConvexHull{N, S1<:LazySet{N}, S2<:LazySet{N}} <: LazySet{N}
     X::S1
     Y::S2
 
     # default constructor with dimension check
-    function ConvexHull(X::S1, Y::S2) where {N<:Real, S1<:LazySet{N},
+    function ConvexHull(X::S1, Y::S2) where {N, S1<:LazySet{N},
```

Comment:
In Julia v1.0 this causes an error:
```julia
Unreachable reached at 0x7f8ac74e4628
signal (4): Illegal instruction
in expression starting at /home/travis/build/JuliaReach/LazySets.jl/test/unit_ConvexHull.jl:1
```
Note that the tests first run `Intersection` but there you kept the annotation.

The error would be fixed like this, and this might actually be better because it does not require the same types:
```julia
function ConvexHull(X::S1, Y::S2) where {S1<:LazySet, S2<:LazySet}
    N = promote_type(eltype(X), eltype(Y))
```

Reply:
great that you found this! i made the change.

Reply:
I guess it is required in all the binary lazy operations unfortunately.

Reply:
ok, i see what happened.. removing the `N` altogether was not intentional here, i just wanted to remove `N<:Real` as far as this PR is concerned.

i agree that it would be nice to create lazy set operations with different numeric types. however, i'm not sure whether changing the inner constructor is a good idea, we may have to actually convert the data to the promoted type, and maybe such generalized outer constructor is better.

Reply:
Here is another version that works:
```julia
function ConvexHull(X::LazySet{N}, Y::LazySet{N}) where {N}
    @assert dim(X) == dim(Y) "sets in a convex hull must have the same dimension"
    return new{N, typeof(X), typeof(Y)}(X, Y)
end
```
`typeof` can always be inferred and hence is the same wrt. run time.
```julia
julia> B = rand(BallInf);

julia> @btime ConvexHull($B, $B)
  19.871 ns (0 allocations: 0 bytes) # this branch
  19.698 ns (0 allocations: 0 bytes) # proposal
```

Reply:
It worked, but needs to be changed for `CartesianProduct`, `Intersection`, and `MinkowskiSum` as well :sweat_smile:

Reply:
Sorry, forgot about the `UnionSet` :disappointed: